### ISSUE TITLE
[native-image] When two substitutions are present the error is not helpful

### ISSUE THREAD

**Issue by graemerocher** [oracle/graal#1715]
Currently `register` looks like:

```
   private static <T> void register(Map<T, T> substitutions, T annotated, T original, T target) {
        if (annotated != null) {
            guarantee(!substitutions.containsKey(annotated) || substitutions.get(annotated) == original || substitutions.get(annotated) == target, "Already registered: %s", annotated);
            substitutions.put(annotated, target);
        }
        ...
    }
```

The output message only indicates the substitution is already registered but not what is causing the conflict. This means the only way for the user to find out is to attach a debugger and set a breakpoint at this line.

The error message should include more information about the two substitutions that conflict so that the user can potentially take appropriate action to remove one of them.

**Comment by viktorklang** [comment]
Seconded. I'm currently having issues trying to track down which of my dependencies is trying to add a substitution to something which clashes somehow. Headache.

**Comment by cstancu** [comment]
This seems like a good candidate for a community contribution.

**Comment by fniephaus** [comment]
@rhyadav I believe this was already addressed in #3238 or is there still more to do here?
